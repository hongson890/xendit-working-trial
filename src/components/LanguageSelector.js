import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';

import { Button, Menu, MenuItem } from '@material-ui/core';
import { IconFlagDE, IconFlagUK } from 'material-ui-flags';
import PublicIcon from '@material-ui/icons/Public';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const btnStyle = {
  marginRight: '16px'
};

export default function LanguageSelector() {
  const { i18n } = useTranslation();

  const [anchorEl, setAnchorEl] = useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const changeLanguage = lng => {
    i18n.changeLanguage(lng);
    // localStorage.setItem('lang', lng);
  };

  return (
    <div className="LanguageSelector">
      <Button
        style={btnStyle}
        endIcon={<ExpandMoreIcon style={{ color: '#7C7C7C' }} />}
        onClick={handleClick}
      >
        <PublicIcon />
      </Button>
      <Menu
        id="language-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem
          onClick={() => {
            handleClose();
            changeLanguage('en');
          }}
        >
          <IconFlagUK />
        </MenuItem>

        <MenuItem
          onClick={() => {
            handleClose();
            changeLanguage('de');
          }}
        >
          <IconFlagDE />
        </MenuItem>
      </Menu>
      {/* <Button style={btnStyle} onClick={() => changeLanguage('en')}><IconFlagUK /></Button>
      <Button onClick={() => changeLanguage('de')}><IconFlagDE /></Button> */}
    </div>
  );
}
