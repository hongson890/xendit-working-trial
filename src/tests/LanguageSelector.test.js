import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import LanguageSelector from '../components/LanguageSelector';

const { i18n } = {};

describe('LanguageSelector', () => {
  it('render language ok', () => {
    const component = shallow(<LanguageSelector />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
