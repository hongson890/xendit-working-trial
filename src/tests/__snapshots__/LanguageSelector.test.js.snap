// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector render language ok 1`] = `
<div
  className="LanguageSelector"
>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    style={Object {}}
  >
    <Component />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    <Component />
  </WithStyles(ForwardRef(Button))>
</div>
`;

exports[`LanguageSelector render ok 1`] = `
<div
  className="LanguageSelector"
>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    style={Object {}}
  >
    <Component />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    <Component />
  </WithStyles(ForwardRef(Button))>
</div>
`;

exports[`LanguageSelector works ok 1`] = `
<div
  className="LanguageSelector"
>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    style={Object {}}
  >
    <Component />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    <Component />
  </WithStyles(ForwardRef(Button))>
</div>
`;
